File: scripts/analyze_radar_h5.py
import h5py
import numpy as np
import sys
import os
import json

radar_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'tmp_radar')
h5_files = [f for f in os.listdir(radar_dir) if f.endswith('.h5') and 'IPRN20' in f]

if not h5_files:
    print("No IPRN20 H5 files found")
    sys.exit(1)

h5_file = os.path.join(radar_dir, sorted(h5_files)[0])
print(f"Analyzing: {os.path.basename(h5_file)}")

with h5py.File(h5_file, 'r') as f:
    print("\n=== ROOT ATTRIBUTES ===")
    for key in f.attrs:
        val = f.attrs[key]
        if isinstance(val, bytes):
            val = val.decode('utf-8', errors='replace')
        print(f"  {key} = {val}")
    
    print("\n=== FILE STRUCTURE ===")
    def visit(name, obj):
        indent = "  " * name.count('/')
        if isinstance(obj, h5py.Group):
            print(f"{indent}GROUP: /{name}")
            for k in obj.attrs:
                v = obj.attrs[k]
                if isinstance(v, bytes):
                    v = v.decode('utf-8', errors='replace')
                elif isinstance(v, np.ndarray):
                    v = v.tolist()
                print(f"{indent}  ATTR: {k} = {str(v)[:150]}")
        elif isinstance(obj, h5py.Dataset):
            print(f"{indent}DATASET: /{name}  shape={obj.shape}  dtype={obj.dtype}")
            for k in obj.attrs:
                v = obj.attrs[k]
                if isinstance(v, bytes):
                    v = v.decode('utf-8', errors='replace')
                elif isinstance(v, np.ndarray):
                    v = v.tolist()
                print(f"{indent}  ATTR: {k} = {str(v)[:150]}")
            # Show sample data for small or large datasets
            if len(obj.shape) > 0:
                data = obj[()]
                if data.size > 0:
                    flat = data.flatten()
                    print(f"{indent}  SAMPLE: {flat[:10].tolist()}")
                    print(f"{indent}  MIN={np.nanmin(data):.4f}  MAX={np.nanmax(data):.4f}  MEAN={np.nanmean(data):.4f}")
    
    f.visititems(visit)

File: scripts/analyze_radar_h5.mjs
import h5wasm from 'h5wasm';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const rootDir = path.join(__dirname, '..');

async function analyze() {
    await h5wasm.ready;

    const dir = path.join(rootDir, 'tmp_radar');
    const files = fs.readdirSync(dir).filter(f => f.endsWith('.h5'));

    if (files.length === 0) { console.log('No H5 files found'); return; }

    const filePath = path.join(dir, files[0]);
    console.log('Analyzing:', files[0]);

    const f = new h5wasm.File(filePath, 'r');

    function listItems(group, prefix = '', depth = 0) {
        const keys = group.keys();
        for (const key of keys) {
            const item = group.get(key);
            const fullPath = prefix + '/' + key;
            const indent = '  '.repeat(depth);

            if (item.type === 'Group') {
                console.log(indent + 'GROUP: ' + fullPath);
                // List attributes
                try {
                    const attrKeys = item.attrs ? Object.keys(item.attrs) : [];
                    for (const aKey of attrKeys) {
                        try {
                            const val = item.attrs[aKey];
                            const valObj = val.value !== undefined ? val.value : val;
                            const valStr = typeof valObj === 'object' ? JSON.stringify(valObj).substring(0, 150) : String(valObj).substring(0, 150);
                            console.log(indent + '  ATTR: ' + aKey + ' = ' + valStr);
                        } catch (e) { console.log(indent + '  ATTR: ' + aKey + ' (error reading)'); }
                    }
                } catch (e) { }
                listItems(item, fullPath, depth + 1);
            } else if (item.type === 'Dataset') {
                console.log(indent + 'DATASET: ' + fullPath + ' shape=' + JSON.stringify(item.shape) + ' dtype=' + item.dtype);
                // List attributes
                try {
                    const attrKeys = item.attrs ? Object.keys(item.attrs) : [];
                    for (const aKey of attrKeys) {
                        try {
                            const val = item.attrs[aKey];
                            const valObj = val.value !== undefined ? val.value : val;
                            const valStr = typeof valObj === 'object' ? JSON.stringify(valObj).substring(0, 150) : String(valObj).substring(0, 150);
                            console.log(indent + '  ATTR: ' + aKey + ' = ' + valStr);
                        } catch (e) { console.log(indent + '  ATTR: ' + aKey + ' (error reading)'); }
                    }
                } catch (e) { }

                // Print a small slice of data if numeric
                try {
                    if (item.shape && item.shape.length > 0) {
                        const slice = item.value;
                        if (slice && slice.length > 0) {
                            // Just show first 10 values
                            const sample = Array.from(slice.slice(0, 10));
                            console.log(indent + '  SAMPLE DATA: ' + JSON.stringify(sample));
                            // Show min/max
                            let min = Infinity, max = -Infinity;
                            for (let i = 0; i < slice.length; i++) {
                                if (slice[i] < min) min = slice[i];
                                if (slice[i] > max) max = slice[i];
                            }
                            console.log(indent + '  MIN=' + min + ' MAX=' + max);
                        }
                    }
                } catch (e) { console.log(indent + '  (cannot read data: ' + e.message + ')'); }
            }
        }
    }

    // Root attributes
    console.log('=== ROOT ATTRIBUTES ===');
    try {
        const rootAttrKeys = f.attrs ? Object.keys(f.attrs) : [];
        for (const key of rootAttrKeys) {
            try {
                const val = f.attrs[key];
                const valObj = val.value !== undefined ? val.value : val;
                const valStr = typeof valObj === 'object' ? JSON.stringify(valObj).substring(0, 200) : String(valObj).substring(0, 200);
                console.log('  ' + key + ' = ' + valStr);
            } catch (e) { console.log('  ' + key + ' (error)'); }
        }
    } catch (e) { console.log('Error reading root attrs:', e.message); }

    console.log('\n=== FILE STRUCTURE ===');
    listItems(f, '', 0);

    f.close();
}

analyze().catch(e => console.error('Error:', e));
